discovery-provider/src/tasks/tracks.py

```diff
@@ -273,6 +371,7 @@ def parse_track_event(
             creator_node_endpoint
         )
 
+        update_track_routes_table(session, track_record, track_metadata)
```

Comment:
Note: In order to use the difference between metdata title and existing record title, we need to check these before the metadata updates the record :)